### PR TITLE
Makes looting/attacking SSD players more obvious

### PR DESCRIPTION
**Pull request by Kyep**
This PR attempts to reduce the problem of people looting from SSDs.

Changes made:
1) Ensures admins still see attack notices against mobs which have no client, BUT are logged_out players. (code\__HELPERS\mobs.dm)
2) Ensures that these attack logs, when seen by admins, have a nice red '(SSD!)' in them next to the name of the SSD player. (code\defines\procs\admin.dm, code\modules\mob\mob_helpers.dm)
~~3) Ensures that when one player goes to strip an item from a SSD player, they get the message: "Warning: looting from SSD players (like [theirname]) is forbidden unless you have ahelped first and got permission." (code\modules\mob\living\living.dm)~~ This bit was taken out because Fox did not like it.
4) Ensures SSD players have "suffering from Space Sleep Disorder" in their on-examine description. Those exact words, to prevent any possible confusion about what kinds of "sleeping" count as SSD. (code/modules/mob/living/carbon/human/examine.dm
)

:cl: Kyep
rscadd: Admins can now see attacks against SSD players more easily.
tweak: Clarified the text you get when examining a SSD player.
/:cl:

### CODE REVIEW

File: code/modules/mob/living/carbon/human/examine.dm [hunk 1]
```diff
@@ -270,7 +270,9 @@
 		msg += "[t_He] [t_has] a stupid expression on [t_his] face.\n"
 
 	if(species.show_ssd && (!species.has_organ["brain"] || get_int_organ(/obj/item/organ/internal/brain)) && stat != DEAD)
-		if(istype(src, /mob/living/carbon/human/interactive))
+		if(src.player_logged)
+			msg += "<span class='deadsay'>[t_He] [t_is] fast asleep, suffering from Space Sleep Disorder.</span>\n"
```
Comment (Fox-McCloud) [inline code/modules/mob/living/carbon/human/examine.dm]: this is already in examine. It doesn't need to be in twice.

**Comment by Kyep** [comment]
@Fox-McCloud I can change the "else if(!key)" or the "else if(!client)" to have this text instead. That's actually what I did originally. Which would you prefer? I still think it would be helpful if the text explicitly said "Space Sleep Disorder" by name.

**Comment by Twinmold93** [comment]
Yes to adding to logs the SSD tags to those that are attack logs during SSD, since that means we don't have to pull up the logs every time someone strips someone of something when they were SSD to verify when they went SSD. No. 3 could be very spammy (they are trying to strip someone of gear, after all). No. 4 is redundent. Only need it in one place, though defined as SSD could be useful to most players.

**Comment by Fox-McCloud** [comment]
:-1: This is an administrative issue, and the codebase isn't the place for dealing with administrative things--it's for having tools to help admins do and perform their jobs, but not for telling players how they should behave or what they should or should not do (from an administrative standpoint).

It's well defined in the server rules that looting SSD's is against the server rules--if someone can't be bothered to read the server rules, that's on them.

This is akin to someone pressing detonate on a toxins bomb and getting the notification "hey, this is against the rules unless you're an antag or using this for legitimate purposes". 

Stuff like this has no place in the codebase; there's already logs when people strip items for others, and when they are stripped, it logs from whom and whether or not that person has a ckey---all the information an admin needs.

**Comment by Kyep** [comment]
I think 1 and 2 should count as "tools to help admins do and perform their jobs".
3 and 4 are just convenience/clarity.

If you want 3 taken out, it can be done without affecting the PR much.

**Comment by Fox-McCloud** [comment]
3 is largely what I take issue with, because it's specifically telling the player how to behave in an administrative capacity, when that's not a code issue.

It's also putting administration into the codebase when downstreams may or may not enforce (or even care) about the SSD rule.

**Comment by Kyep** [comment]
Alright then, 3 deleted. PR updated.

What do you say now, Fox?

**Comment by Fox-McCloud** [comment]
That's fine; the rest are perfectly valid tools for admins to complete/do their jobs.

File: code/modules/mob/mob_helpers.dm [hunk 2]
```diff
@@ -34,6 +34,13 @@
 			return 1
 	return 0
 
+proc/isSSD(A)
+	if(istype(A,/mob))
+		var/mob/M = A
+		if (M.player_logged)
```
Comment (Fox-McCloud) [inline code/modules/mob/mob_helpers.dm]: *glare*

File: code/modules/mob/mob_helpers.dm [hunk 3]
```diff
@@ -34,6 +34,13 @@
 			return 1
 	return 0
 
+proc/isSSD(A)
+	if(istype(A,/mob))
```
Comment (Fox-McCloud) [inline code/modules/mob/mob_helpers.dm]: `if(istype(A, /mob))`

File: code/modules/mob/mob_helpers.dm [hunk 4]
```diff
@@ -34,6 +34,13 @@
 			return 1
 	return 0
 
+proc/isSSD(A)
+	if(istype(A, /mob))
+		var/mob/M = A
+		if(M.player_logged)
+			return 1
+	return 0
```
Comment (Krausus) [inline code/modules/mob/mob_helpers.dm]: This whole proc can be simplified to:
```
/proc/isSSD(mob/M)
	return istype(M) && M.player_logged
```

**Comment by Earthdivine** [comment]
Yes please :+1:

**Comment by Lady-Luck** [comment]
>3) Ensures that when one player goes to strip an item from a SSD player, they get the message: "Warning: looting from SSD players (like [theirname]) is forbidden unless you have ahelped first and got permission." (code\modules\mob\living\living.dm) 

Had a thought about this. Why not make it so the message is a config option that's disabled by default?

**Pull request merged**